railties/test/backtrace_cleaner_test.rb

```diff
@@ -8,6 +8,13 @@ def setup
     @cleaner = Rails::BacktraceCleaner.new
   end
 
+  test "should filter '\tfrom' before processing" do
+    backtrace = ["\tfrom /Path/to/railsapp/app/models/user.rb:22:in `bar'"]
+    result = @cleaner.clean(backtrace, :all)
+    assert_equal "/Path/to/railsapp/app/models/user.rb:22:in `bar'", result[0]
+    assert_equal 1, result.length
+  end
```

Comment:
I think we need to be creating real backtraces in these tests. My thinking is that this bug showed up because the format of the backtrace changed but all of these tests are looking at hardcoded backtraces.

For example, I just created a fresh app on Ruby 2.7.6 and the backtrace looks different than the one on 3.1 (it starts with a number):

```
irb(main):001> User.new.foo
Traceback (most recent call last):
        ... 1 levels...
        30: from bin/rails:4:in `require'
        29: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/commands.rb:18:in `<top (required)>'
        28: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/command.rb:69:in `invoke'
        27: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/command.rb:149:in `with_argv'
        26: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/command.rb:71:in `block in invoke'
        25: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/command/base.rb:73:in `perform'
        24: from /home/hartley/.cache/asdf/installs/ruby/2.7.6/lib/ruby/gems/2.7.0/gems/thor-1.2.2/lib/thor.rb:392:in `dispatch'
        23: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/command/base.rb:178:in `invoke_command'
        22: from /home/hartley/.cache/asdf/installs/ruby/2.7.6/lib/ruby/gems/2.7.0/gems/thor-1.2.2/lib/thor/invocation.rb:127:in `invoke_command'
        21: from /home/hartley/.cache/asdf/installs/ruby/2.7.6/lib/ruby/gems/2.7.0/gems/thor-1.2.2/lib/thor/command.rb:27:in `run'
        20: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/commands/console/console_command.rb:107:in `perform'
        19: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/commands/console/console_command.rb:17:in `start'
        18: from /home/hartley/src/github.com/skipkayhil/rails/railties/lib/rails/commands/console/console_command.rb:79:in `start'
         3: from (irb):1:in `<main>'
         2: from /home/hartley/test/dev_minimal/app/models/user.rb:4:in `foo'
         1: from /home/hartley/test/dev_minimal/app/models/user.rb:8:in `bar'
/home/hartley/test/dev_minimal/app/models/user.rb:14:in `baz': unhandled exception
```

Reply:
What you're describing here is a [fixture](https://guides.rubyonrails.org/testing.html#the-low-down-on-fixtures) a well-defined and well-supported approach to testing in rails.  Having hard coded backtraces is no different than a fixture.

Again, not saying this is a good idea or perfect solution.  But I do know this PR fixes the immediate issue with modern ruby versions and modern rails versions and does so following the existing not-perfect pattern.